presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/type/ShortTimestampEncoder.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.base.type;
+
+import io.prestosql.spi.block.BlockBuilder;
+import io.prestosql.spi.type.TimestampType;
+import org.joda.time.DateTimeZone;
+
+import static io.prestosql.plugin.base.type.PrestoTimestampEncoderFactory.scaleEpochMillisToMicros;
+import static io.prestosql.spi.type.Timestamps.MICROSECONDS_PER_SECOND;
+import static io.prestosql.spi.type.Timestamps.MILLISECONDS_PER_SECOND;
+import static io.prestosql.spi.type.Timestamps.NANOSECONDS_PER_MICROSECOND;
+import static io.prestosql.spi.type.Timestamps.round;
+
+class ShortTimestampEncoder
+        extends AbstractPrestoTimestampEncoder<Long>
+{
+    ShortTimestampEncoder(TimestampType type, DateTimeZone timeZone)
+    {
+        super(type, timeZone);
+    }
+
+    @Override
+    public Long write(Long micros, BlockBuilder blockBuilder)
+    {
+        type.writeLong(blockBuilder, micros);
+        return micros;
+    }
+
+    @Override
+    public Long getTimestamp(DecodedTimestamp decodedTimestamp)
+    {
+        long micros;
+        if (timeZone != DateTimeZone.UTC) {
+            micros = scaleEpochMillisToMicros(timeZone.convertUTCToLocal(decodedTimestamp.getEpochSeconds() * MILLISECONDS_PER_SECOND));
```

Comment:
inline `scaleEpochMillisToMicros`. The only value of that method is doing checked multiply -- which is nice, but we do not do this in other places (eg 2 lines below)

Reply:
No, I'll add some.

Reply:
@aalbu let's use unchecked multiply now
my reasoning is -- the problem is general and pre-existing (like here https://github.com/prestosql/presto/blob/be2b97a8480f657f311e673fa38867c8faf4e2df/presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/StandardColumnMappings.java#L416)
we need to solve the problem holistically.
fixing in one place doesn't buy us much
and fixing in all the placesa the at same time is an opportunity to define useful abstractions

I logged https://github.com/prestosql/presto/issues/5168 to track this